common/src/main/java/org/apache/drill/common/expression/SchemaPath.java

```diff
@@ -220,9 +214,6 @@ public boolean contains(Object obj) {
     }
 
     SchemaPath other = (SchemaPath) obj;
-    if (rootSegment == null) {
```

Comment:
One can indeed pass a null rootSegment to the constructor. If this is something that shouldn't happen, maybe I can just add an assertion in the constructors, otherwise I will revert back the change in this function